server/src/test/java/org/elasticsearch/index/engine/NoOpEngineTests.java

```diff
@@ -168,49 +166,28 @@ public void testTrimUnreferencedTranslogFiles() throws Exception {
         tracker.updateFromMaster(1L, Collections.singleton(allocationId.getId()), table);
         tracker.activatePrimaryMode(SequenceNumbers.NO_OPS_PERFORMED);
 
-        boolean softDeleteEnabled = engine.config().getIndexSettings().isSoftDeleteEnabled();
         final int numDocs = scaledRandomIntBetween(10, 3000);
         for (int i = 0; i < numDocs; i++) {
             engine.index(indexForDoc(createParsedDoc(Integer.toString(i), null)));
             tracker.updateLocalCheckpoint(allocationId.getId(), i);
             if (rarely()) {
                 engine.flush();
             }
+            if (randomBoolean()) {
+                engine.rollTranslogGeneration();
+            }
         }
+        // prevent translog from trimming so we can test trimUnreferencedFiles in NoOpEngine.
+        final Translog.Snapshot snapshot = engine.getTranslog().newSnapshot();
         engine.flush(true, true);
-
-        final String translogUuid = engine.getTranslog().getTranslogUUID();
-        final long minFileGeneration = engine.getTranslog().getMinFileGeneration();
-        final long currentFileGeneration = engine.getTranslog().currentFileGeneration();
         engine.close();
 
         final NoOpEngine noOpEngine = new NoOpEngine(noOpConfig(INDEX_SETTINGS, store, primaryTranslogDir, tracker));
-        final Path translogPath = noOpEngine.config().getTranslogConfig().getTranslogPath();
-
-        final long lastCommitedTranslogGeneration;
-        try (Engine.IndexCommitRef indexCommitRef = noOpEngine.acquireLastIndexCommit(false)) {
-            Map<String, String> lastCommittedUserData = indexCommitRef.getIndexCommit().getUserData();
-            lastCommitedTranslogGeneration = Long.parseLong(lastCommittedUserData.get(Translog.TRANSLOG_GENERATION_KEY));
-            assertThat(lastCommitedTranslogGeneration, equalTo(currentFileGeneration));
-        }
-
-        assertThat(Translog.readMinTranslogGeneration(translogPath, translogUuid), equalTo(minFileGeneration));
-        assertThat(noOpEngine.getTranslogStats().estimatedNumberOfOperations(), equalTo(softDeleteEnabled ? 0 : numDocs));
-        assertThat(noOpEngine.getTranslogStats().getUncommittedOperations(), equalTo(0));
-
         noOpEngine.trimUnreferencedTranslogFiles();
```

Comment:
Should we check prior to this that we have some operations to trim, except in the case where the last thing done was a flush?

Reply:
++. Added in https://github.com/elastic/elasticsearch/pull/51905/commits/02103b77d611ab3fc10d0d90299f28c7508b9cd8.